openlibrary/catalog/add_book/__init__.py

```diff
@@ -908,13 +911,15 @@ def update_work_with_rec_data(
     """
     # Add subjects to work, if not already present
     if 'subjects' in rec:
-        work_subjects = list(work.get('subjects', []))
-        for s in rec['subjects']:
-            if s not in work_subjects:
-                work_subjects.append(s)
-                need_work_save = True
-        if need_work_save and work_subjects:
-            work['subjects'] = work_subjects
+        work_subjects: list[str] = list(work.get('subjects', []))
+        rec_subjects: list[str] = rec.get('subjects', [])
+        deduped_subjects = uniq(
+            itertools.chain(work_subjects, rec_subjects), lambda item: item.casefold()
+        )
+
+        if len(work_subjects) != len(deduped_subjects):
```

Comment:
This isn't sufficient ; there might be cases where the deduping might remove a pre-existing subject, but the import adds a subject, resulting in the lists being the same length!

```suggestion
        if work_subjects != deduped_subjects:
```

Reply:
Arrgggg! Thank you for catching this. I updated the tests so they won't pass with this terrible `len()` based comparison. :D